### PR TITLE
feat: implement process digital identity anchor

### PR DESCRIPTION
**Pull request by namhoang1604**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds the processDigitalIdentityAnchor function

## Related Tickets & Documents

https://github.com/gs-gs/fa-ag-trace/issues/632

## Mobile & Desktop Screenshots/Recordings

image

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [x] 📕 Documentation
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: documentation/docs/mock-apps/services/index.md [hunk 1]
```diff
@@ -37,68 +40,63 @@ graph TD
...
-                            }
-                        }
-                    },
-                    "identifierKeyPath": "/product/itemIdentifiers/0/identifierValue"
-                }
-            ]
-        },
+          "vckit": {
+            "vckitAPIUrl": "http://localhost:3332/v2",
+            "issuer": "did:web:agtrace-passports.s3.ap-southeast-2.amazonaws.com"
```
Comment (ldhyen99) [inline documentation/docs/mock-apps/services/index.md:59]: Should we change this issuer to an example?

Comment (namhoang1604) [reply documentation/docs/mock-apps/services/index.md]: Updated :)

**Review by ldhyen99 — DISMISSED** [review]

File: documentation/docs/mock-apps/services/process-digital-identity-anchor.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+---
+sidebar_position: 27
```
Comment (huynguyen-hl) [inline documentation/docs/mock-apps/services/process-digital-identity-anchor.md:2]: I think this `sidebar_position: 27` might overlap with the `Merge To Local Storage` document's sidebar_position. Could you double check it?”

File: packages/services/src/validateContext.ts [hunk 3]
```diff
@@ -60,6 +62,7 @@ export const validateContextTransformationEvent = (
   if (_.isEmpty(context.vckit.vckitAPIUrl)) return error('Invalid vckitAPIUrl');
   if (_.isEmpty(context.vckit.issuer)) return error('Invalid issuer');
 
+  if (_.isEmpty(context.dpp)) return error('Invalid dpp context');
   if (_.isEmpty(context.dpp.context)) return error('Invalid dpp context');
   if (_.isEmpty(context.dpp.type)) return error('Invalid dpp type');
   if (_.isEmpty(context.dpp.dlrLinkTitle)) return error('Invalid dpp dlrLinkTitle');
```
Comment (huynguyen-hl) [inline packages/services/src/validateContext.ts:68]: I think the `validateContextTransformationEvent` function should include a check/validate to ensure that the `context.dppCredentials` value exists.

File: packages/services/src/processDigitalIdentityAnchor.service.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    vcKitAPIUrl: vckit.vckitAPIUrl,
+    issuer: vckit.issuer,
+    context: digitalIdentityAnchor.context,
+    type: digitalIdentityAnchor.type,
+    restOfVC: {
+      render: digitalIdentityAnchor.renderTemplate,
+    },
+  });
+
+  const vcUrl = await getStorageServiceLink(storage, vc, `${identifier}/${generateUUID()}`);
```
Comment (huynguyen-hl) [inline packages/services/src/processDigitalIdentityAnchor.service.ts:47]: I have a bit of concern about the `getStorageServiceLink` function name. If this function primarily focuses on uploading data to an object storage service and returning the link to the uploaded object, should we renaming it to make its purpose clearer?

For example, we could rename it to something like `uploadVC`or `uploadObject` to better reflect its purpose. I think so.

Comment (namhoang1604) [reply packages/services/src/processDigitalIdentityAnchor.service.ts:47]: Agree that the name of the function does not describe the main functionality. Let's update it in another PR

File: documentation/docs/mock-apps/services/process-digital-identity-anchor.md (hunk 2 above)
Comment (namhoang1604) [reply documentation/docs/mock-apps/services/process-digital-identity-anchor.md]: Good catch!

**Review by huynguyen-hl — APPROVED** [review]

**Review by ldhyen99 — APPROVED** [review]

**Pull request merged**